build-tools/packages/build-cli/src/library/repoPolicyCheck/fluidBuildDatabase.ts

```diff
@@ -0,0 +1,373 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import path from "node:path";
+
+import { type Package, TscUtils } from "@fluidframework/build-tools";
+import type { TsConfigJson } from "type-fest";
+
+import { ApiLevel } from "../apiLevel";
+import { ApiTag } from "../apiTag";
+import { queryOutputMapsFromPackageExports } from "../packageExports";
+
+type PackageName = string;
+type Script = string;
+/**
+ * `${package.name}#${script}`
+ */
+export type PackageNameAndScript = string;
+type AbsoluteFilePath = string;
+type ModuleType = "CommonJS" | "ESM";
+/**
+ * Representation of a package.json script that generates "build" output.
+ */
+export interface BuildScript {
+	/**
+	 * Name of the package where script is located.
+	 */
+	packageName: PackageName;
+	/**
+	 * Name of the script within package.
+	 */
+	script: Script;
+	/**
+	 * The recognized module type, if any, that the script generates.
+	 * An `undefined` value indicates that script is not known to be associated
+	 * with a certain module type. This might happen for script that only copies
+	 * files as copying files has no inherent association to one module type or
+	 * the other.
+	 */
+	moduleType: ModuleType | undefined;
+}
+
+function readArgValues<TQuery extends Record<string, string>>(
+	commandLine: string,
+	argQuery: TQuery,
+): TQuery {
+	const values: Record<string, string> = {};
+	const args = commandLine.split(" ");
+	for (const [argName, defaultValue] of Object.entries(argQuery)) {
+		const indexOfArgValue = args.indexOf(`--${argName}`) + 1;
+		values[argName] =
+			0 < indexOfArgValue && indexOfArgValue < args.length
+				? args[indexOfArgValue]
+				: defaultValue;
+	}
+	return values as TQuery;
+}
+
+function flubOutput(
+	pkg: Package,
+	commandLine: string,
+): { files: AbsoluteFilePath[]; type: ModuleType | undefined } | undefined {
+	if (!commandLine.startsWith("flub generate entrypoints")) {
```

Comment:
Is this check because this function runs as a part of `flub generate entrypoints`? This feels fragile to me. At the very least, it seems like we should extract this into a constant with some docs.

Reply:
This is not a part of `flub generate entrypoints`. This is helper to derive what output a `flub` command would generate. `generate entrypoints` is the only one I know we need.
It would be better if *Task classes or some part of them could be leveraged. I checked that out briefly but there are too many dependencies and complications.
Same for the fluid commands. Maybe if I understood oclif better there would be a better refactor to reuse the command processing. There is also a complication with code organization and cyclical import dependencies that make relocating Commands hard. Best I could do was extract some common logic. Having said that I might actually be able to clear up the cycle and do something better.

Reply:
Relocated command and shared more logic